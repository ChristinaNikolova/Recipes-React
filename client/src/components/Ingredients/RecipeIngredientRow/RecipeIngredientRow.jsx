import toastr from 'toastr';

import * as ingredientsService from '../../../services/ingredientsService.js';

function RecipeIngredientRow({ ingredientId, recipeId, ingredientName, quantity, clickHandler }) {
    const remove = () => {
        ingredientsService
            .remove(recipeId, ingredientId)
            .then((data) => {
                if (data['status'] === 400) {
                    toastr.error(data['message'], 'Error');
                    return;
                }

                toastr.success(data['message'], 'Success');
                clickHandler();
            });
    }

    return (
        <tr>
            <td>{ingredientName}</td>
            <td>{quantity}</td>
            <td><button className="special-btn danger" onClick={remove}>Delete</button></td>
        </tr >
    );
}

export default RecipeIngredientRow;