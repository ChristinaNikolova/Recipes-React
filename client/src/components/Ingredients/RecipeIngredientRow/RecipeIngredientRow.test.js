import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RecipeIngredientRow from './RecipeIngredientRow';

it('should render RecipeIngredientRow component', () => {
    const wrapper = shallow(<RecipeIngredientRow />);
    expect(toJson(wrapper)).toMatchSnapshot();
});