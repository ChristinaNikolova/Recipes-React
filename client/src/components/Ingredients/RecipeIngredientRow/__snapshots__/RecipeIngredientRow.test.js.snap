// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RecipeIngredientRow component 1`] = `
<tr>
  <td />
  <td />
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Delete
    </button>
  </td>
</tr>
`;
